### ISSUE TITLE
Login loop

### ISSUE THREAD

**Issue by Jooho** [fabric8io/openshift-auth-proxy#18]
Hi, 

I encounter to login kinbana loop issue but I don't see detail  log.
 
If I go to kibana web console through openshift master console, I got this error:
~~
in passport.ensureAuthenticated for req path /auth/token
not authenticated by request session.
in validateBearerToken

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: SELF_SIGNED_CERT_IN_CHAIN
    at SecurePair.<anonymous> (tls.js:1381:32)
    at SecurePair.emit (events.js:92:17)
    at SecurePair.maybeInitFinished (tls.js:980:10)
    at CleartextStream.read [as _read] (tls.js:472:13)
    at CleartextStream.Readable.read (_stream_readable.js:341:10)
    at EncryptedStream.write [as _write] (tls.js:369:25)
    at doWrite (_stream_writable.js:226:10)
    at writeOrBuffer (_stream_writable.js:216:5)
    at EncryptedStream.Writable.write (_stream_writable.js:183:11)
    at write (_stream_readable.js:602:24)
    at flow (_stream_readable.js:611:7)
    at Socket.pipeOnReadable (_stream_readable.js:643:5)
~~

If I go to kibana web console directly, I can see this:
~~~
Starting up the proxy with auth mode "oauth2" and proxy transform "user_header,token_header".
in passport.ensureAuthenticated for req path /
not authenticated by request session.
x.x.x.x - - [22/Jun/2017:19:48:13 +0000] "GET / HTTP/1.1" 302 0 "-" "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"
in callback handler for req path /auth/openshift/callback
x.x.x.x - - [22/Jun/2017:19:48:20 +0000] "GET /auth/openshift/callback?code=FozCl0AjJyGwht_Mllo8OJ4_9aNDNmBLUBp1LqlQmvk&state= HTTP/1.1" 302 58 "https://ocp-np-master.iad.ca.inet:8443/login?.....
~~~

What does SELF_SIGNED_CERT_IN_CHAIN mean? what should I take a look at more?
 
Environment:
  Openshift v3.3.1
  Image registry.access.redhat.com/openshift3/logging-kibana:3.3.1

**Comment by jcantrill** [comment]
I would consider moving to a version of Openshift logging for 3.4 or better which is the 2.4.x line of Elasticsearch.  We have seen this issue in a few places but please provide additional information: https://github.com/openshift/origin-aggregated-logging/blob/master/docs/issues.md

**Comment by Jooho** [comment]
@jcantrill 

I can reproduce this issue on my end so I update the information that you know from my test machine.
This is the information:

OpenShift Cluster Details

Cluster version  OCP v3.3.1
oc v3.3.1.38
kubernetes v1.3.0+52492b4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://master-cluster.example.com:8443
openshift v3.3.1.38
kubernetes v1.3.0+52492b4

Cluster infrastructure KVM
Number of infrastructure nodes and size  2Core 4G 
Number of compute nodes and size  2Core/4G
Logging Image versions  3.3.1
ADD NOTE HERE ON HOW TO RETRIEVE IMAGE VERSION FROM OCP AND ORIGIN

Installer Details

How was logging installed :   both Openshift Ansible | Deployer
Openshift installer version
Ansible inventory file (attatched)
Ansible installer logs
Logging Component Details

MAYBE WE CAN GRAB SOME OF THESE BY LABEL


Output of https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py

ERROR==> 

$python ./cron-send-logging-checks.py
Traceback (most recent call last):
  File "./cron-send-logging-checks.py", line 31, in <module>
    from openshift_tools.monitoring.ocutil import OCUtil
ImportError: No module named openshift_tools.monitoring.ocutil



Moreover, I created this bugzilla :https://bugzilla.redhat.com/show_bug.cgi?id=1464215

[test-hosts.txt](https://github.com/fabric8io/openshift-auth-proxy/files/1116041/test-hosts.txt)

**Comment by jcantrill** [comment]
I would encourage you to move to 3.4 or later of the logging stack as we are unlikely to fix anything related to 3.3.  Closing this issue in favor of the BZ
